### PR TITLE
Group nodes

### PR DESCRIPTION
**Pull request by SHKnudsen**
### Purpose

This PR adds a new Group class with 9 nodes to interact with Revit Groups through Dynamo

AllNodes

All of the nodes have related unit and system tests, which has been run in RTF before submitting.

#### Unit Test
GroupNodes_UnitTest_210120

#### System Test
GroupNodes_SystemTest_210120

### Declarations
Check these if you believe they are true

- [x] The code base is in a better state after this PR
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [x] The level of testing this PR includes is appropriate
- [x] User facing strings, if any, are extracted into `*.resx` files
- [x] Snapshot of UI changes, if any.

### Reviewers
@AndyDu1985
@radumg
@mjkkirschner 
@ZiyunShang 

### FYIs
Screenshots of the dynamo graphs for the system test
#### Group.ByElements
CanCreateGroupFromListOfElements

#### Group.GetMembers
CanGetGroupMembers

#### Group Properties
CanGetGroupProperties

#### Group.PlaceInstance
CanPlaceGroupInstance

#### Group.UngroupElements
CanUngroupElementsFromGroup

### CODE REVIEW

File: src/Libraries/RevitNodes/Elements/Group.cs [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+
+            return newGroup;
+        }
+
+        #endregion
+
+        #region public properties
+
+        /// <summary>
+        ///
```
Comment (AndyDu1985) [inline src/Libraries/RevitNodes/Elements/Group.cs]: Why empty here?

File: src/Libraries/RevitNodes/Elements/Group.cs [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+                if (this.InternalGroup.IsAttached)
+                    throw new InvalidOperationException(Properties.Resources.AttachedGroupLocation);
+
+                var locationPt = this.InternalGroup.Location as LocationPoint;
+                return locationPt.Point.ToPoint();
+            }
+        }
+
+        /// <summary>
+        ///
```
Comment (AndyDu1985) [inline src/Libraries/RevitNodes/Elements/Group.cs]: Same as above.

**Comment by ZiyunShang** [comment]
Hi @SHKnudsen , because of ElementType nodes merged, your PR may have conflicts with master branch, could you resolve them?

File: src/Libraries/RevitNodes/Elements/Group.cs [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+                return attachedGroupIds.Select(id => Document.GetElement(id).ToDSType(true)).ToList();
+            }
+        }
+
+        #endregion
+
+        #region public methods
+
+        /// <summary>
+        ///
```
Comment (ZiyunShang) [inline src/Libraries/RevitNodes/Elements/Group.cs]: I think here should add some descriptions.

File: src/Libraries/RevitNodes/Elements/Group.cs [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+        public List<Element> GetMembers()
+        {
+            TransactionManager.Instance.EnsureInTransaction(Document);
+            List<ElementId> memberIds = this.InternalGroup.GetMemberIds().ToList();
+            TransactionManager.Instance.TransactionTaskDone();
+            return memberIds.Select(id => Document.GetElement(id).ToDSType(true)).ToList();         
+        }
+
+        /// <summary>
+        ///
```
Comment (ZiyunShang) [inline src/Libraries/RevitNodes/Elements/Group.cs]: Here should add some descriptions too.

**Pull request merged**